doc/user/mpi-io.md

```diff
@@ -0,0 +1,93 @@
+# MPI-IO Support
+
+The Message Passing Interface (MPI) Standard, 
+maintained by the [MPI Forum](https://www.mpi-forum.org/docs/),
+includes a chapter on MPI-IO. 
+
+[ROMIO](https://www.mcs.anl.gov/projects/romio/) is a well-known
+implementation of MPI-IO and is included in many MPI implementations.
+
+DAOS provides its own MPI-IO ROMIO ADIO driver, located in a MPICH fork on
+the [DAOS-Stack GitHub](https://github.com/daos-stack/mpich). 
```

Comment:
I think at this point we can just point to the upstream repo.
The internal repo is just being used for our CI testing and doesn't have anything additional.

Reply:
let's fix this in a subsequent PR.